externals/CMakeLists.txt

```diff
@@ -123,3 +123,9 @@ macro(ProcessWithHeaderize headervar sourcevar inputfile outdir)
     list(APPEND ${headervar} ${outputfile}.hpp)
     list(APPEND ${sourcevar} ${outputfile}.cpp)
 endmacro()
+
+# Catch
+set(NO_SELFTEST ON CACHE BOOL "We trust that the testing library does not need to be tested." FORCE)
```

Comment:
If the library is header only, you shouldn't have to add_subdirectory (and thus not set any of its options).